src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Workspace.cs

```diff
@@ -552,4 +545,28 @@ private async Task WhenInitialized(CancellationToken cancellationToken)
             Verify.NotDisposed(this);
         }
     }
+
+    /// <summary>
+    /// Maps from our property data snapshot to Roslyn's API for accessing the project's
+    /// evaluation data.
+    /// </summary>
+    private sealed class EvaluationDataAdapter : EvaluationData
+    {
+        private readonly IImmutableDictionary<string, string> _properties;
+
+        public EvaluationDataAdapter(IImmutableDictionary<string, string> properties)
+        {
+            _properties = properties;
+        }
+
+        public override string GetPropertyValue(string name)
+        {
+            _properties.TryGetValue(name, out string? value);
+
+            // Return the empty string rather than null.
+            value ??= "";
```

Comment:
>"" [](http://example.com/codeflow?start=22&length=2)

Unless something has changed in the compiler, it is preferable to use String.Empty here to avoid allocating another string object for "".

Reply:
Thankfully the compiler hasn't done that for a long time.